### DISCUSSION TITLE
Is there a way to have the "Home" & "End" keys close the popup and go to the start (home) or end of the prompt (normal functionality w/o the extention)

### DISCUSSION THREAD

**Discussion by CCpt5** [DominikDoom/a1111-sd-webui-tagcomplete#139] [General]
Hey,

First off thanks for an exceptional extension - it's by far one of the most useful for me w/ A1111. 

I'm curious if there is an option, or a way to keep the "Home" / "End" keyboard keys from interactacting w/ the popup auto-opened, but instead close the popup and go to the start or end of the prompt (per usual). Perhaps I have an odd workflow but I find myself getting "stuck" in the popup selection box a lot since by nature I use those 2 keys to navigate quickly to the start/end of the prompt (but don't ever use them w/in the pop-up selection)  
 
If this is something that either isn't possible due to gradio or contrasts with intended functionality - all good! I've just gotten caught in the popup enough due to my use of those keys that I figured I'd inquire.

Thanks for any insight!

-C

**Comment by DominikDoom** [comment]
There is indeed a way for just that, it was added pretty recently for #127 with a very similar reason to yours.

I haven't gotten to document it properly in the readme yet, but in the extensions's tags folder there should be a "keymap.json" file where you can set different navigation hotkeys. You can set the JumpToStart / End bindings there to a different key or "" / "None" to disable them. In which case home and end will then close the pop-up and jump in the text just like you want.

**Reply by brknsoul** [reply]
> but in the extensions's tags folder there should be a "keymap.json"

I don't see this file. Could you knock up an example of how it should look?

Nevermind, it's in the Settings;
![image](https://github.com/DominikDoom/a1111-sd-webui-tagcomplete/assets/1969381/d30f76e1-c7ee-4d10-be23-0170df2c2317)

**Reply by DominikDoom** [reply]
Yeah, everything that was in separate json files before was moved to the settings page a few months ago to prevent upstream changes from overwriting local edits people have made.

**Comment by CCpt5** [comment]
I will give this a try. Thank you so much!!
